### PR TITLE
In-app Log Monitoring

### PR DESCRIPTION
**Pull request by a3957273**

### CODE REVIEW

**Comment by ARADDCC002** [comment]
I think the search button on the admin log filters should be moved to be inside the menu

File: pages/admin.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    search: '',
+    isRegex: false,
+  })
+
+  const handleGetLogs = (filters: LogFilters): void => {
+    setLogFilters(filters)
+  }
+
+  return (
+    <Wrapper title='Admin' page='admin'>
```
Comment (ARADDCC002) [inline pages/admin.tsx:25]: We should check the user role here as well and return an error message. Right now if you view the page as a non-admin you just get the log box with a "loading..." text.

File: server/routes/v1/admin.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+export const getApplicationLogs = [
+  ensureUserRole(['user', 'admin']),
+  async (req: Request, res: Response) => {
+    const after = parseDateQuery(req.query.after, new Date(0))
+    const before = parseDateQuery(req.query.before, new Date(Date.now()))
+    const level = parseInt(parseString('level', req.query.level) || '0', 10)
+
+    if (Number.isNaN(level)) {
+      throw BadReq({ level: req.query.level }, 'Level cannot be passed as number')
```
Comment (ARADDCC002) [inline server/routes/v1/admin.ts:67]: Would parsed make more sense here?

Comment (ARADDCC012) [reply server/routes/v1/admin.ts:67]: I think that's what the intention was. I'll update under that assumption.

File: pages/admin.tsx (hunk 1 above)
Comment (ARADDCC012) [reply pages/admin.tsx:26]: Good catch, I'll put some generic unauthorised message in the event that someone manually navigates to the page.

File: server/models/Version.ts [hunk 3]
```diff
@@ -1,8 +1,14 @@
 import { Document, IndexOptions, model, Schema, Types } from 'mongoose'
 import { LogStatement } from './Deployment'
-import logger from '../utils/logger'
 import { approvalStateOptions, ApprovalStates, DateString } from '../../types/interfaces'
 import { ModelDoc } from './Model'
+import logger from '../utils/logger'
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function doNotUse() {
```
[the remark this answers is not in the document]
Comment (ARADDCC012) [reply server/models/Version.ts]: I've reimplemented the use of `logger` in `VersionSchema.log`. If that's not something we want then we might need to see why it's required for Version.

**Pull request merged**